### PR TITLE
[dua] fix Parent failed to send reregister for MTD Child on receiving `ST_DUA_REREGISTER`

### PR DESCRIPTION
**Pull request by simonlingoogle**
This commit enhances DUA response handling for `ST_DUA_REREGISTER`:
- Parent will stop registering DUA for MTD Child until the next `Child Update Request`
- MTD Child will re-register its DUA by `Child Update Request` or `DUA.req`

### CODE REVIEW

File: src/core/thread/dua_manager.cpp [hunk 1]
```diff
@@ -476,7 +476,7 @@ void DuaManager::PerformNextRegistration(void)
         const Ip6::Address *duaPtr = nullptr;
         Child *             child  = nullptr;
 
-        if (!mRegisterCurrentChildIndex)
+        if (!mRegisterCurrentChildIndex || !mChildDuaMask.Get(mChildIndexDuaRegistering))
```
Comment (librasungirl) [inline src/core/thread/dua_manager.cpp:479]: keep it as is and reset `mRegisterCurrentChildIndex` to false in below ?https://github.com/openthread/openthread/pull/6637/files#diff-f22621e2122bf5e36a030cbcbfbe861983e321e66d3e38921632626b9eb3b47aR664

`mRegisterCurrentChildIndex` is designed to directly reuse the `mChildIndexDuaRegistering`  and skip the unnecessary iteration below if we know the index to register already.

Comment (simonlingoogle) [reply src/core/thread/dua_manager.cpp]: `mRegisterCurrentChildIndex` related logic is now removed.

Comment (librasungirl) [reply src/core/thread/dua_manager.cpp]: Thanks for cleaning up this. 👍

File: src/core/thread/dua_manager.cpp [hunk 2]
```diff
@@ -623,8 +621,11 @@ Error DuaManager::ProcessDuaResponse(Coap::Message &aMessage)
             mDuaState             = kRegistered;
             break;
         case ThreadStatusTlv::kDuaReRegister:
-            mDuaState                  = kToRegister;
-            mDelay.mFields.mCheckDelay = Mle::kImmediateReRegisterDelay;
+            if (Get<ThreadNetif>().HasUnicastAddress(GetDomainUnicastAddress()))
+            {
+                RemoveDomainUnicastAddress();
+                AddDomainUnicastAddress();
```
Comment (simonlingoogle) [inline src/core/thread/dua_manager.cpp:627]: Explain: Removing and adding the DUA again triggers both DUA.req and `Child Update Request` (for MTD).

**Review by librasungirl — APPROVED** [review]
👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**